### PR TITLE
Fixes #83 Sharing images via a list of apps

### LINKED ISSUE #83: Allowing share via multiple platforms
**Actual Behavior**
Currently, the app allows photos to be shared explicitly via Bluetooth.

**Expected Behavior**
We can extend it to a 1 click share button that allows sharing via any app that allows Image content. 

**Would you like to work on the issue?**

Yes, I would like to work on it.

### PR DESCRIPTION
**Pull request by harsu-ag**
Fixes issue #83

Changes: Bluetooth share was changed to share intent. In the process, the Bluetooth activity invocation was completely removed. 

Also, the targetSdk was changed to 23, for sdk 24+ we need FileProviders for generating URI that in turn doesn't support RemovableStorage sharing. 
This change will not affect any of the code's working.

### CODE REVIEW

File: app/src/main/java/vn/mbm/phimp/me/image/ImageSharer.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+ * Created by harshit on 07-03-2017.
+ */
+
+public class ImageSharer {
+    public static void shareSingle(Context context, String filePath) {
+        Intent share = new Intent(Intent.ACTION_SEND);
+        share.setType("image/*");
+        File imageFileToShare = new File(filePath);
+        Uri imageURI = Uri.fromFile(imageFileToShare);
+//                FileProvider.getUriForFile(context, BuildConfig.APPLICATION_ID + ".provider", imageFileToShare);
```
Comment (devkhan) [inline app/src/main/java/vn/mbm/phimp/me/image/ImageSharer.java]: Remove code if not needed.

**Comment by harsu-ag** [comment]
@devkhan done. 
Also removed conflicts in build.gradle file

**Comment by Abhi2424shek** [comment]
Again conflicts!

**Comment by harsu-ag** [comment]
Resolved

**Comment by Abhi2424shek** [comment]
Pls do open multiple PRs for multiple issues. I.e, 1 PR for 1 Issue

**Comment by harsu-ag** [comment]
@Abhi2424shek when I committed my changes to my development branch, It got added to this PR. 
I forgot to create a new branch for the same. 

How can I correct this.

**Comment by CloudyPadmal** [comment]
@harsu-ag you can reset the head and create a new branch and cherry-pick the commit you want to send a seperate PR :)

> $ git checkout -b seperatePR origin/master
> $ git cherry-pick <hash of the commit>
> $ git push origin seperatePR

Hope this helps :metal:

**Comment by harsu-ag** [comment]
Thanks @CloudyPadmal . Done @Abhi2424shek

File: app/src/main/java/vn/mbm/phimp/me/image/ImageSharer.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+import java.io.File;
+import java.util.ArrayList;
+
+/**
+ * Created by harshit on 07-03-2017.
+ */
+
+public class ImageSharer {
+    public static void shareSingle(Context context, String filePath) {
```
Comment (Abhi2424shek) [inline app/src/main/java/vn/mbm/phimp/me/image/ImageSharer.java]: Can you reduce the two functions for sharing into a single function?

Comment (harsu-ag) [inline app/src/main/java/vn/mbm/phimp/me/image/ImageSharer.java]: Sure. :)

Comment (harsu-ag) [inline app/src/main/java/vn/mbm/phimp/me/image/ImageSharer.java]: @Abhi2424shek done.

File: app/src/main/java/vn/mbm/phimp/me/Upload.java [hunk 3]
```diff
@@ -257,23 +259,24 @@ public void onItemClick(AdapterView<?> parent, View v, int position, long id) {
...
                     intent.setClass(getContext(), BluetoothShareMultipleFile.class);
                     intent.putExtra("imagelist", imagelist);
                     intent.putExtra("activityName", "Upload");
                     intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
-                    startActivity(intent);
+                    startActivity(intent);*/
 
                 } else {
-                    Commons.AlertLog(ctx, "Do not have photo to share", getString(R.string.accept)).show();
+                    Commons.AlertLog(ctx, "Do not have photo to shareSingle", getString(R.string.accept)).show();
```
Comment (Abhi2424shek) [inline app/src/main/java/vn/mbm/phimp/me/Upload.java]: shareSingle?

Comment (harsu-ag) [inline app/src/main/java/vn/mbm/phimp/me/Upload.java]: My refractor probably replaced all strings also along with variable names. :(

Comment (harsu-ag) [inline app/src/main/java/vn/mbm/phimp/me/Upload.java]: @Abhi2424shek  should I remove the previous code that I have commented?

Comment (Abhi2424shek) [inline app/src/main/java/vn/mbm/phimp/me/Upload.java]: Remove the previous code you have commented, make the logs and comments  logic. (Besure about Strings) :) @harsu-ag

**Comment by harsu-ag** [comment]
@Abhi2424shek Done. :)

**Comment by Abhi2424shek** [comment]
Resolve the conflicts pls.

**Comment by Abhi2424shek** [comment]
@harsu-ag pls resolve the conflicts pls and update the upload_apk_forks.sh file see the contributing.md file pls.

**Comment by pa1pal** [comment]
@harsu-ag are you working on it? No updates in this PR. Other people are up for this issue.
Also don't downgrade the targetSDK version, add the support possible in latest version. Will figure out other solution in future.

**Comment by Abhi2424shek** [comment]
@pa1pal I am afraid we don't have option to get URIs of images in sdcard in nouget. I think that's why he downgraded.

**Comment by Abhi2424shek** [comment]
Closing the PR because of inactivity.

**Pull request closed without merging**